### ISSUE TITLE
Port twisted.internet.test.test_tls to Python 3

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6172]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)| @itamarst reported|
|-|-|
|Trac ID|trac#6172|
|Type|enhancement|
|Created|2012-10-31 15:21:43Z|
|Branch|https://github.com/twisted/twisted/tree/tlstests-py3-6172|

`twisted.internet.test.test_tls` should run on Python 3. This depends on [#6157](https://github.com/twisted/twisted/issues/6157).

<details><summary>Searchable metadata</summary>

```
trac-id__6172 6172
type__enhancement enhancement
reporter__itamar itamar
priority__normal normal
milestone__Python_3_3_Minimal Python 3.3 Minimal
branch__branches_tlstests_py3_6172 branches/tlstests-py3-6172
branch_author__itamarst itamarst
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1351696903000000 1351696903000000
changetime__1352299033000000 1352299033000000
version__None None
owner__itamar itamar

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_1"></a>|@itamarst commented|
|-|-|

(In [36355]) Branching to 'tlstests-py3-6172'

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_2"></a>|@itamarst set owner to @exarkun|
|-|-|

http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/tlstests-py3-6172 is running. Another minimal port; one more ticket and SSL is ported.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun set owner to @itamarst|
|-|-|

Great.  Please merge.

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_4"></a>|@itamarst set status to `closed`|
|-|-|

(In [36359]) Merge tlstests-py3-6172.

Author: itamar
Review: exarkun
Fixes: [#6172](https://github.com/twisted/twisted/issues/6172)

Port twisted.internet.test.test_tls to Python 3.
